WalletWasabi/Hwi/HwiProcessManager.cs

```diff
@@ -0,0 +1,245 @@
+﻿using NBitcoin;
+using NBitcoin.BIP174;
+using Newtonsoft.Json.Linq;
+using Nito.AsyncEx;
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Linq;
+using System.Runtime.InteropServices;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using WalletWasabi.Helpers;
+using WalletWasabi.Hwi.Models;
+using WalletWasabi.Logging;
+
+namespace WalletWasabi.Hwi
+{
+	/// <summary>
+	/// https://github.com/bitcoin-core/HWI
+	/// </summary>
+	public static class HwiProcessManager
+	{
+		public static string HwiPath { get; private set; }
+		public static AsyncLock AsyncLock { get; }
+		public static Network Network { get; private set; }
+
+		static HwiProcessManager()
+		{
+			AsyncLock = new AsyncLock();
+			Network = Network.Main;
+		}
+
+		public static async Task<PSBT> SignTxAsync(HardwareWalletInfo hardwareWalletInfo, PSBT psbt)
+		{
+			var psbtString = psbt.ToString();
+			var networkString = Network == Network.Main ? "" : "--testnet ";
+			JToken jtok = await SendCommandAsync($"{networkString}--device-type \"{hardwareWalletInfo.Type.ToString().ToLowerInvariant()}\" --device-path \"{hardwareWalletInfo.Path}\" signtx {psbtString}");
+			JObject json = jtok as JObject;
+			var signedPsbtString = json.Value<string>("psbt");
+			var signedPsbt = PSBT.Parse(signedPsbtString);
+			signedPsbt.Finalize();
+
+			return signedPsbt;
+		}
+
+		/// <summary>
+		/// Acquire bech32 xpub on path: m/84h/0h/0h
+		/// https://github.com/bitcoin-core/HWI/blob/master/docs/examples.md
+		/// </summary>
+		public static async Task<ExtPubKey> GetXpubAsync(HardwareWalletInfo hardwareWalletInfo)
+		{
+			var networkString = Network == Network.Main ? "" : "--testnet ";
+			JToken jtok = await SendCommandAsync($"{networkString}--device-type \"{hardwareWalletInfo.Type.ToString().ToLowerInvariant()}\" --device-path \"{hardwareWalletInfo.Path}\" getxpub m/84h/0h/0h");
```

Comment:
This line doesn't work on my Linux.
It generates:
```
$ ./hwi --testnet --device-type \"coldcard\" --device-path \"/tmp/ckcc-simulator.sock\" getxpub m/84h/0h/0h
{"error": "Unknown device type specified", "code": -4}
```

However the following line works okay:
```
$ ./hwi --testnet --device-type coldcard getxpub m/84h/0h/0h
{"xpub":"tpubDCx8y86cKonoPyTtj3f9NZLpBYoBNkbAzUdafMHhggjxkhF8Dny2aekWfDafywEMZEQaQjkK9Gxn7aN7usLRUQdYbvDgcnmYRf72khPEouL"}
```

![image](https://user-images.githubusercontent.com/127973/56150110-75e87080-5f84-11e9-871e-52b54992054b.png)

Reply:
That'd explain why I couldn't make it work on my VM.

Reply:
Character escapeout? Anyway I'll get back to it after I succeed to send a transaction :)

Reply:
No it doesn't. Lucas, remove the excapes and try then. When I try with cmd or bash I got "open failed", but when I try with Wasabi code on Windows it succeeds. But when I try with Wasabi code on Linux I get "open failed".  

If I try without removing the `\` characters before the `"` characters, then I get the error you get.

Reply:
`open failed` on Linux usually means the udev rules aren't installed. I've made a PR to HWI that has all of the udev rules and instructions for installing them: https://github.com/bitcoin-core/HWI/pull/147

Reply:
In the Windows cli if I replace double `\` with single ones then it'll work properly as it does in the software. So I think your Linux issue will be solved if you remove these `\` too.

Reply:
Oh! Sorry, you are right here (I didn't see my error before). These two lines work equally well:
```
$ ./hwi --testnet --device-type "coldcard" --device-path "/tmp/ckcc-simulator.sock" getxpub m/84h/0h/0h
{"xpub":"tpubDCx8y86cKonoPyTtj3f9NZLpBYoBNkbAzUdafMHhggjxkhF8Dny2aekWfDafywEMZEQaQjkK9Gxn7aN7usLRUQdYbvDgcnmYRf72khPEouL"}

$ ./hwi --testnet --device-type coldcard --device-path /tmp/ckcc-simulator.sock getxpub m/84h/0h/0h
{"xpub":"tpubDCx8y86cKonoPyTtj3f9NZLpBYoBNkbAzUdafMHhggjxkhF8Dny2aekWfDafywEMZEQaQjkK9Gxn7aN7usLRUQdYbvDgcnmYRf72khPEouL"}
```

Reply:
It still doesn't work on my Linux VM nor on my Linux real machine. Neither in Wasabi nor with Terminal.

Reply:
Oh I missed @achow101's reply. Thanks, that should solve it. Will report back.

Reply:
We should ship the "udev rules" if we can.

Reply:
https://github.com/zkSNACKs/WalletWasabi/issues/1343